sqlmesh/utils/jinja.py

```diff
@@ -228,9 +228,24 @@ def extract_macro_references_and_variables(
                         )
                     )
 
+        def is_variable_node(n: nodes.Node) -> bool:
+            return (
+                isinstance(n, nodes.Call)
+                and isinstance(n.node, nodes.Name)
+                and n.node.name in (c.VAR, c.BLUEPRINT_VAR)
+            )
```

Comment:
I'd extract this helper out instead of making it a closure to reduce the complexity of avoid increasing the complexity of this extractor.